diff -urN grub-0.97/grub/Makefile.in grub-0.97-patched/grub/Makefile.in
--- a/grub/Makefile.in	2005-05-08 06:42:34.000000000 +0400
+++ b/grub/Makefile.in	2011-11-30 17:46:45.000000000 +0400
@@ -84,6 +84,7 @@
 CCASFLAGS = @CCASFLAGS@
 CCDEPMODE = @CCDEPMODE@
 CFLAGS = @CFLAGS@
+override CFLAGS:=$(subst nocona,i486,$(subst -m64,-fno-stack-protector,$(CFLAGS)))
 CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
 CYGPATH_W = @CYGPATH_W@
diff -urN grub-0.97/stage2/Makefile.in grub-0.97-patched/stage2/Makefile.in
--- a/stage2/Makefile.in	2005-05-08 06:42:39.000000000 +0400
+++ b/stage2/Makefile.in	2011-11-30 17:42:25.000000000 +0400
@@ -331,6 +331,7 @@
 CCASFLAGS = @CCASFLAGS@
 CCDEPMODE = @CCDEPMODE@
 CFLAGS = @CFLAGS@
+override CFLAGS:=$(subst nocona,i486,$(subst -m64,-fno-stack-protector,$(CFLAGS)))
 CPP = @CPP@
 CPPFLAGS = @CPPFLAGS@
 CYGPATH_W = @CYGPATH_W@
