diff -urN astor2-einarc-svn1882/src/raid/software.rb astor2-einarc-svn1882-usb-filtering/src/raid/software.rb
--- astor2-einarc-svn1882/src/raid/software.rb	2011-03-25 14:28:07.000000000 +0300
+++ astor2-einarc-svn1882-usb-filtering/src/raid/software.rb	2011-05-30 16:35:25.000000000 +0400
@@ -293,7 +293,7 @@
 
 			for device in devices
 				# Possibility to skip USB mass storage devices
-				# next if usb_device?(device)
+				next if usb_device?(device)
 				target = phys_to_scsi(device.gsub(/^\/dev\//, ''))
 				d = { :state => 'unknown' }
 				d[:vendor] = physical_read_file(device, "device/vendor") or ""
@@ -459,7 +459,7 @@
 
 		def usb_device?(device)
 			name = device.gsub(/^\/dev\//, '')
-			return (not [File.read("/sys/block/#{name}/uevent")].grep(/usb/).empty?)
+			return File.readlink( "/sys/block/#{name} ) =~ /usb/
 		end
 
 		def active?(device)
