diff -urN astor2-einarc-svn1962/src/raid/software.rb astor2-einarc-svn1962-usb-filtering/src/raid/software.rb
--- astor2-einarc-svn1962/src/raid/software.rb	2011-10-27 20:54:46.000000000 +0400
+++ astor2-einarc-svn1962-usb-filtering/src/raid/software.rb	2011-10-29 00:39:57.000000000 +0400
@@ -296,7 +296,7 @@
 
 			for device in devices
 				# Possibility to skip USB mass storage devices
-				# next if usb_device?(device)
+				next if usb_device?(device)
 				target = phys_to_scsi(device.gsub(/^\/dev\//, ''))
 				d = { :state => 'unknown' }
 				d[:vendor] = physical_read_file(device, "device/vendor") or ""
@@ -566,6 +566,10 @@
 
 		def usb_device?(device)
 			name = device.gsub(/^\/dev\//, '')
+
+			File.read("/proc/cmdline") =~ /root=\/dev\/(...)/
+			return true if $1 == name
+
 			return ((not File.readlines("/sys/block/#{name}/uevent").grep(/usb/).empty?) or
 			       (File.readlink("/sys/block/#{name}" ) =~ /usb/))
 		end
