diff -urN astor2-einarc-svn1968/src/raid/software.rb astor2-einarc-svn1968-forced-assembling/src/raid/software.rb
--- astor2-einarc-svn1968/src/raid/software.rb	2012-01-24 12:14:51.000000000 +0400
+++ astor2-einarc-svn1968-forced-assembling/src/raid/software.rb	2012-01-24 12:15:55.000000000 +0400
@@ -51,7 +51,7 @@
 				name, uuid = $1, $2
 				next unless ( name and uuid ) # mdadm 3.x can print arrays without any /dev entry
 				name.gsub!( "md/", "md" )
-				run("--assemble --uuid=#{uuid} #{name}") unless active?(name)
+				run("--assemble --force --uuid=#{uuid} #{name}") unless active?(name)
 			end
 		end
 		
